controllers/component_build_controller.go

```diff
@@ -543,8 +549,37 @@ func ConfigureRepositoryForPaC(component appstudiov1alpha1.Component, config map
 		return prUrl, nil
 
 	case "gitlab":
-		// TODO implement
-		return "", fmt.Errorf("git provider %s is not supported", gitProvider)
+		glclient, err := gitlab.NewGitlabClient(accessToken)
+		if err != nil {
+			return "", err
+		}
+
+		gitlabNamespace := gitSourceUrlParts[3]
+		gitlabProjectName := gitSourceUrlParts[4]
+		projectPath := gitlabNamespace + "/" + gitlabProjectName
+
+		err = gitlab.SetupPaCWebhook(glclient, projectPath, webhookTargetUrl, webhookSecret)
+		if err != nil {
+			return "", err
+		}
```

Comment:
I'm not concerned about testing for errors when getting the client, but is there any validation of the inputs into `gitlab.SetupPaCWebhook(glclient, projectPath, webhookTargetUrl, webhookSecret)` that make sense (and maybe add unit tests around those?

Reply:
where we might provide betterr error information to the user than the libraries we are calling?

Reply:
> is there any validation of the inputs into gitlab.SetupPaCWebhook(glclient, projectPath, webhookTargetUrl, webhookSecret)

`glclient` is created above. The only argument it gets is the access token.
`projectPath` is built from the repository url. If the url is wrong we'll get 404 later.
`webhookTargetUrl` is read from the Openshift cluster
`webhookSecret` is generated by our code

To sum up the above: we might want to have a better validation of user data (token and git url specifically) by querying something from the project just to check if everything works. What do you think?
Anyway, I'd like to leave that implementation for later, if any.

> where we might provide betterr error information to the user than the libraries we are calling?

I am not sure I got the question correctly, but my plan for the future is to have wrapper/custom error and just log/create event for it.

Reply:
> > is there any validation of the inputs into gitlab.SetupPaCWebhook(glclient, projectPath, webhookTargetUrl, webhookSecret)
> 
> `glclient` is created above. The only argument it gets is the access token. `projectPath` is built from the repository url. If the url is wrong we'll get 404 later. `webhookTargetUrl` is read from the Openshift cluster `webhookSecret` is generated by our code
> 
> To sum up the above: we might want to have a better validation of user data (token and git url specifically) by querying something from the project just to check if everything works. What do you think? Anyway, I'd like to leave that implementation for later, if any.

yep for what it is worth openshift builds does some git ls-remote calls (much less expensive then git clone) to validate git related inputs and catch errors as soon as possible, so there is precedence for such a validation, where that validation is proven useful

I'm fine if you want to leave that for later.  Either add it to your existing jira task around error handling, or open a similar but separate item.

Feel free to mark this item resolved once you've captured things in a Jira.

> 
> > where we might provide betterr error information to the user than the libraries we are calling?
> 
> I am not sure I got the question correctly, but my plan for the future is to have wrapper/custom error and just log/create event for it.

that sounds ^^ exactly around the path I was envisioning